### ISSUE TITLE
Proper domain setup with vercel

### ISSUE THREAD

**Issue by PatrickManfra** [vercel/commerce#1218]
I am noticing an error with my domain. I would like vercel to host my app, but I am getting the same error that others are getting with their sitemap when you go to build the production version on vercel.

When I verify my domain with shopify I get my production url stuck on this, but the app works on my vercel domain just fine.
<img width="887" alt="Screenshot 2023-10-11 at 3 20 54 PM" src="https://github.com/vercel/commerce/assets/44379983/d779eadc-3a29-4634-9ff4-65839be19416">

When i have the domain configured with Vercel I get the sitemap error. I do not change anything else. Which means that I can not use this method to host my store after building a whole site in NextJs for it. 

The project builds fine locally but does not build properly on the production URL. I think this is a common problem people are facing. The setup guide talks nothing about DNS configuration or settings what so ever. 

Basically wasted a few days getting this to work, and I think its not fixed as prior stated in closed issues. 

Could someone from the team please help me out?

**Comment by leerob** [comment]
Hey! It's hard to tell what specifically is the issue here, but I'd recommend running through this again https://vercel.com/docs/integrations/shopify.

**Comment by ericjiji3** [comment]
I am having the same issue right now as well!

**Comment by manovotny** [comment]
Hey all! If you're experiencing sitemap build errors, it's likely that you do not have your environment variables setup correctly in the Vercel dashboard for your project. The sitemap fails to build because it cannot call Shopify to get the data it needs to create the sitemap.

https://vercel.com/docs/integrations/shopify#configure-environment-variables

**Comment by ericjiji3** [comment]
Yes, I actually just solved the issue and it was like you said, because of environment variables. In my situation, I accidently set the SHOPIFY_STORE_DOMAIN value to my 3rd party domain rather than the Shopify domain(domainname.myshopify.com)

**Comment by manovotny** [comment]
@ericjiji3 glad to hear you got it resolved. ❤️
